### ISSUE TITLE
Error: Cannot statically analyse 'require(…, …)' in line 16

### ISSUE THREAD

**Issue by wscourge** [jonschlinkert/markdown-toc#173]
See: https://stackoverflow.com/questions/69898244/cannot-statically-analyse-require-in-line-16

**Problem:**

When importing markdown-toc in nextjs I am getting the issue. /pages/index.js

import toc from "markdown-toc";
Then I got the below error

./node_modules/markdown-toc/lib/utils.js
Cannot statically analyse 'require(…, …)' in line 16

**Why:**

`require` gets overwritten by the utils assignment here:

```
'use strict';

/**
 * Module dependencies
 */

var diacritics = require('diacritics-map');
var utils = require('lazy-cache')(require);
var fn = require;
require = utils;

/**
 * Lazily required module dependencies
 */

require('concat-stream', 'concat');
```

**Comment by tinesoft** [comment]
Any update on this?

I'm having  the same error when importing the library in an Angular 14 application...

**Comment by wscourge** [comment]
I don't believe there will be, there are > 50 open issues and no update in the last 5 years.

**Comment by Antonio225t** [comment]
This fixed the problem but maybe you have to mess with dependencies...

Copy the library and put it in another folder to edit the source of it, then remove this:
```
var fn = require;
require = utils;
```
And replace all the `require(..., ...)` with `utils(..., ...)` like this:
```
utils('concat-stream', 'concat');
utils('gray-matter', 'matter');
utils('list-item', 'li');
utils('markdown-link', 'mdlink');
utils('minimist');
utils('mixin-deep', 'merge');
utils('object.pick', 'pick');
utils('remarkable', 'Remarkable');
utils('repeat-string', 'repeat');
utils('strip-color');
```

Hope it helps!

**Comment by Antonio225t** [comment]
Ok, I think I have a better solution. Try this: "markdown-toc-unlazy" https://www.npmjs.com/package/markdown-toc-unlazy
